sentry/src/routes/campaign.rs

```diff
@@ -1,56 +1,286 @@
+        return Err(ResponseError::Forbidden("Request not sent by campaign creator".to_string()))
+    }
 
-    // insert Campaign
+    let error_response = ResponseError::BadRequest("err occurred; please try again later".to_string());
+
+    let accounting_spent = get_accounting_spent(app.pool.clone(), &campaign.creator, &campaign.channel.id()).await?;
+
+    // TODO: AIP#61: Update when changes to Spendable are ready
+    let latest_spendable = fetch_spendable(app.pool.clone(), &campaign.creator, &campaign.channel.id()).await?;
+    let total_deposited = latest_spendable.deposit.total;
 
-    // match insert_campaign(&app.pool, &campaign).await {
-    //     Err(error) => {
-    //         error!(&app.logger, "{}", &error; "module" => "create_channel");
-
-    //         match error {
-    //             PoolError::Backend(error) if error.code() == Some(&SqlState::UNIQUE_VIOLATION) => {
-    //                 Err(ResponseError::Conflict(
-    //                     "channel already exists".to_string(),
-    //                 ))
-    //             }
-    //             _ => Err(error_response),
-    //         }
-    //     }
-    //     Ok(false) => Err(error_response),
-    //     _ => Ok(()),
-    // }?;
-
-    let create_response = SuccessResponse { success: true };
+    let remaining_for_channel = total_deposited.checked_sub(&accounting_spent).ok_or(ResponseError::FailedValidation("couldn't calculate remaining for channel".to_string()))?;
+
+    if campaign.budget > remaining_for_channel {
+        return Err(ResponseError::BadRequest("Not Enough budget for campaign".to_string()));
+    }
+
+    // If the campaign is being created, the amount spent is 0, therefore remaining = budget
+    set_initial_remaining_for_campaign(&app.redis.clone(), campaign.id, campaign.budget).await.map_err(|_| ResponseError::BadRequest("Couldn't update remaining while creating campaign".to_string()))?;
+
+    // insert Campaign
+    match insert_campaign(&app.pool, &campaign).await {
+        Err(error) => {
+            error!(&app.logger, "{}", &error; "module" => "create_campaign");
+            match error {
+                PoolError::Backend(error) if error.code() == Some(&SqlState::UNIQUE_VIOLATION) => {
+                    Err(ResponseError::Conflict(
+                        "Campaign already exists".to_string(),
+                    ))
+                }
+                _ => Err(error_response),
+            }
+        }
+        Ok(false) => Err(ResponseError::BadRequest("Encountered error while creating Campaign; please try again".to_string())),
+        _ => Ok(()),
+    }?;
 
     Ok(success_response(serde_json::to_string(&campaign)?))
 }
 
+pub mod update_campaign {
+    use super::*;
+    use crate::fetch_campaign;
+
+    pub const CAMPAIGN_REMAINING_KEY: &'static str = "campaignRemaining";
+
+    pub async fn set_initial_remaining_for_campaign(redis: &MultiplexedConnection, id: CampaignId, amount: UnifiedNum) -> Result<bool, Error> {
```

Comment:
Return a value that would be used to check if the Campaign id already existed in Redis. This would be another check if the campaign already exist in combination with the Postgres one.

Reply:
Can you clarify? Wouldn't the `SETNX` return `None` which would therefore return `Error` if it already exists?